sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -61,8 +66,20 @@ def __init__(self, cache_manager=None):
     self._watching_set = set()
     # Holds variables list of (Dict[str, object]).
     self._watching_dict_list = []
+    # Holds results of pipeline runs as Dict[Pipeline, PipelineResult].
+    # Each key is a pipeline instance defined by the end user. The
+    # InteractiveRunner is responsible for populating this dictionary
+    # implicitly.
+    self._pipeline_results = {}
     # Always watch __main__ module.
     self.watch('__main__')
+    # Do a warning level logging if current python version is below 2 or 3.5.3.
+    if sys.version_info < (3,):
+      logging.warning('Interactive Beam does not support Python 2.')
+    elif sys.version_info < (3, 5, 3):
```

Comment:
Should we simply say interactivity required 3.5.3 ?

Reply:
We can. I'll make it simpler. Let's just consider every different component a big feature. The feature requires Py 3.5.3. And we NOOP if dependency is not ready.